search/ui/class-vip-search-widget.php

```diff
@@ -0,0 +1,454 @@
+	 *
+	 * @param array $args     Widgets args supplied by the theme.
+	 * @param array $instance The current widget instance.
+	 *
+	 */
+	public function widget( $args, $instance ) {
+		$instance = $this->vip_search_populate_defaults( $instance );
+
+		if ( empty( $instance['search_box_enabled'] ) && empty( $instance['user_sort_enabled'] ) ) {
+			return;
+		}
+
+		$title = ! empty( $instance['title'] ) ? $instance['title'] : '';
+
+		/** This filter is documented in core/src/wp-includes/default-widgets.php */
+		$title = apply_filters( 'widget_title', $title, $instance, $this->id_base );
+
+		echo $args['before_widget']; //phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		?>
+			<div id="<?php echo esc_attr( $this->id ); ?>-wrapper" >
+		<?php
+
+		if ( ! empty( $title ) ) {
+			$this->render_widget_title( $title, $args['before_title'], $args['after_title'] );
+		}
+
+		$default_sort            = $instance['sort'] ?? self::DEFAULT_SORT;
+		list( $orderby, $order ) = $this->sorting_to_wp_query_param( $default_sort );
+		$current_sort            = "{$orderby}|{$order}";
+
+		// we need to dynamically inject the sort field into the search box when the search box is enabled, and display
+		// it separately when it's not.
+		if ( ! empty( $instance['search_box_enabled'] ) ) {
+			$this->render_widget_search_form( $instance['post_types'], $orderby, $order );
+		}
+
+		if ( ! empty( $instance['search_box_enabled'] ) && ! empty( $instance['user_sort_enabled'] ) ) :
+			?>
+			<div class="vip-search-sort-wrapper">
+				<label>
+					<?php esc_html_e( 'Sort by', 'jetpack' ); ?>
+					<select class="vip-search-sort">
+						<?php foreach ( $this->get_sort_types() as $sort => $label ) { ?>
+							<option value="<?php echo esc_attr( $sort ); ?>" <?php selected( $current_sort, $sort ); ?>>
+								<?php echo esc_html( $label ); ?>
+							</option>
+						<?php } ?>
+					</select>
+				</label>
+			</div>
+			<?php
+		endif;
+
+		$this->maybe_render_sort_javascript( $instance, $order, $orderby );
+
+		echo '</div>';
+		echo $args['after_widget']; //phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	}
+
+
+	/**
+	 * Renders JavaScript for the sorting controls on the frontend.
+	 *
+	 * This JS is a bit complicated, but here's what it's trying to do:
+	 * - find the search form
+	 * - find the orderby/order fields and set default values
+	 * - detect changes to the sort field, if it exists, and use it to set the order field values
+	 *
+	 * @param array  $instance The current widget instance.
+	 * @param string $order    The order to initialize the select with.
+	 * @param string $orderby  The orderby to initialize the select with.
+	 *
+	 */
+	private function maybe_render_sort_javascript( array $instance, string $order, string $orderby ) {
+		if ( ! empty( $instance['user_sort_enabled'] ) ) :
+			?>
+			<script type="text/javascript">
+				var vipSearchModuleSorting = function() {
```

Comment:
What do you think about moving this to a JS file and only printing the required "dynamic" portion of the data and pass that as function arguments?

Reply:
That would be a nice improvement to have, but for now I'd rather stick to Jetpack's implementation.